builtins/target-avx512skx-x64.ll

```diff
@@ -55,7 +55,107 @@ define i1 @__none(<WIDTH x MASK> %mask) nounwind readnone alwaysinline {
+  %high = shufflevector <64 x i16> %input, <64 x i16> undef, <32 x i32> <i32 32, i32 33, i32 34, i32 35, i32 36, i32 37, i32 38, i32 39, i32 40, i32 41, i32 42, i32 43, i32 44, i32 45, i32 46, i32 47, i32 48, i32 49, i32 50, i32 51, i32 52, i32 53, i32 54, i32 55, i32 56, i32 57, i32 58, i32 59, i32 60, i32 61, i32 62, i32 63>
+
+  ; Get first 32 and second 32 indices
+  %perm16 = trunc <64 x i32> %perm to <64 x i16>
+  %perm_low = shufflevector <64 x i16> %perm16, <64 x i16> undef, <32 x i32> <i32 0, i32 1, i32 2, i32 3, i32 4, i32 5, i32 6, i32 7, i32 8, i32 9, i32 10, i32 11, i32 12, i32 13, i32 14, i32 15, i32 16, i32 17, i32 18, i32 19, i32 20, i32 21, i32 22, i32 23, i32 24, i32 25, i32 26, i32 27, i32 28, i32 29, i32 30, i32 31>
+  %perm_high = shufflevector <64 x i16> %perm16, <64 x i16> undef, <32 x i32> <i32 32, i32 33, i32 34, i32 35, i32 36, i32 37, i32 38, i32 39, i32 40, i32 41, i32 42, i32 43, i32 44, i32 45, i32 46, i32 47, i32 48, i32 49, i32 50, i32 51, i32 52, i32 53, i32 54, i32 55, i32 56, i32 57, i32 58, i32 59, i32 60, i32 61, i32 62, i32 63>
+
+  ; Two 512-bit VPERMI2W operations
+  %result1 = call <32 x i16> @llvm.x86.avx512.vpermi2var.hi.512(<32 x i16> %low, <32 x i16> %perm_low, <32 x i16> %high)
+  %result2 = call <32 x i16> @llvm.x86.avx512.vpermi2var.hi.512(<32 x i16> %low, <32 x i16> %perm_high, <32 x i16> %high)
+
+  ; Concatenate results
+  %final = shufflevector <32 x i16> %result1, <32 x i16> %result2, <64 x i32> <i32 0, i32 1, i32 2, i32 3, i32 4, i32 5, i32 6, i32 7, i32 8, i32 9, i32 10, i32 11, i32 12, i32 13, i32 14, i32 15, i32 16, i32 17, i32 18, i32 19, i32 20, i32 21, i32 22, i32 23, i32 24, i32 25, i32 26, i32 27, i32 28, i32 29, i32 30, i32 31, i32 32, i32 33, i32 34, i32 35, i32 36, i32 37, i32 38, i32 39, i32 40, i32 41, i32 42, i32 43, i32 44, i32 45, i32 46, i32 47, i32 48, i32 49, i32 50, i32 51, i32 52, i32 53, i32 54, i32 55, i32 56, i32 57, i32 58, i32 59, i32 60, i32 61, i32 62, i32 63>
+  ret <64 x i16> %final
+}
+
+define <WIDTH x i8> @__shuffle_i8(<WIDTH x i8>, <WIDTH x i32>) nounwind readnone alwaysinline {
+  %vals = zext <WIDTH x i8> %0 to <WIDTH x i16>
+  %res = call <WIDTH x i16> @__shuffle_i16(<WIDTH x i16> %vals, <WIDTH x i32> %1)
+  %res_i8 = trunc <WIDTH x i16> %res to <WIDTH x i8>
+  ret <WIDTH x i8> %res_i8
+}
+
+define <WIDTH x half> @__shuffle_half(<WIDTH x half>, <WIDTH x i32>) nounwind readnone alwaysinline {
+  %vals = bitcast <WIDTH x half> %0 to <WIDTH x i16>
+  %res = call <WIDTH x i16> @__shuffle_i16(<WIDTH x i16> %vals, <WIDTH x i32> %1)
+  %res_half = bitcast <WIDTH x i16> %res to <WIDTH x half>
+  ret <WIDTH x half> %res_half
+}
+
+declare <64 x i32> @llvm.masked.gather.v64i32.v64p0i32(<64 x i32*>, i32, <64 x i1>, <64 x i32>)
```

Comment:
It probably doesn't matter much but in `v64p0i32` do we really need i32?

Reply:
Not really, since pointer are opaque. Fixed.